### PR TITLE
fix(challenges): Rosetta Code test errors and formatting

### LINKED ISSUE #277: Problems with layout and tests for various Rosetta Code challenges
#### Describe the bug
This was brought up on the Contributor's Chat Page by @Dangeranger. Seems that a number of the Rosetta Code challenges have minor/major layout issues, and also problems with the tests. Here's @Dangeranger's screenshot of the tests throwing errors:

image

Seems like this is because some challenges rely on an array of tests named tests in the tail section to fill in parts of the texts for the test suite section of the page. However, because the Curriculum JSON has a field named tests, when a solution is submitted on the page it instead calls the JSON tests array rather than the one from the tail.

The easiest way to fix this is by going through the existing problems, giving the tests array in the tail a unique name and adjusting the layout as needed. For many of the problems I removed the unnecessary paragraph tags. 

Below is a list of challenges I made more major formatting changes to:

* "9 billion names of God the integer": Fixed the number triangle
* "Averages/Pythagorean means": Fixed the example output object
* "ABC Problem": Fixed pre section
* "Align columns": Fixed pre section and lists
* "Closest-pair problem": Fixed pre sections
* "Combinations": Fixed pre section
* "Comma quibbling": Fixed ordered and list
* "Compare a list of strings": Fixed list and i tag
* "Convert seconds to compound duration": Fixed lists and tables
* "Count occurrences of a substring": Fixed list
* "Count the coins": Fixed list
* "Cramer's rule": Fixed math function
* "Date Format": Fixed list and spacing
* "Date manipulation": Made headings bold
* "Deal cards for FreeCell": Fixed lists
* "Deep copy": Fixed lists
* "Department numbers": Fixed lists
* "Execute Brain****": Added a table
* "Element-wise operations": Fixed list
* "Emirp primes": Fixed list
* "Ethiopian multiplication": Fixed pre sections and lists
* "Euler method": Fixed lists
* "Execute a Markov algorithm": Fixed pre sections
* "Factors of a Mersenne number": Fixed spacing and pre section
* "Farey sequence": Rearranged sentences and fixed list
* "Fibonacci n-step number sequences": Added a table
* "Fibonacci word": Fixed pre section
* "Hailstone sequence": Fixed lists
* "Hash join": Fixed pseudo code and added table
* "Sailors, coconuts and a monkey problem": Fixed spacing
* "SEDOLs": Fixed spacing
* "Tokenize a string with escaping": Reformatted description and added code tags
* "Top rank per group": Fixed pre tags
* "Tower of Hanoi": Fixed spacing
* "Vector cross product": Fixed spacing
* "Vector dot product": Fixed spacing
* "Word wrap": Fixed spacing and pre tags
* "Y combinator": Fixed spacing
* "Zeckendorf number representation": Fixed spacing and i tag
* "Zhang-Suen thinning algorithm": Fixed pre tags and spacing
* "Zig-zag matrix": Fixed spacing and pre tags

And challenges that had problems with the tests or challengeType:

* "Date manipulation": Renamed the tests arr and hard coded the tests and answer strings
* "Define a primitive data type": Moved throws function to tail section so the tests work
* "Execute a Markov algorithm": Moved rules, markovTests (renamed from tests), and outputs to tail section
* "Fractran": Changed challenge type to 5, renamed the tests array in the tail, and hard coded the test strings and answers into the individual tests
* "Gamma function": Renamed tests array in tail and hard coded the test strings and answers into the individual tests
* "Gaussian elimination": Renamed tests array in tail and hard coded the test strings and answers into the individual tests
* "General FizzBuzz": Renamed tests array in tail and hard coded the test strings and answers into the individual tests
* "Greatest subsequential sum": Renamed tests array in tail and hard coded the test strings and answers into the individual tests
* "IBAN": Renamed tests array in tail and hard coded the test strings and answers into the individual tests
* "Jaro distance": Renamed tests array in tail and hard coded the test strings and answers into the individual tests
* "JortSort": Renamed tests array in tail and hard coded the test strings and answers into the individual tests

Note that none of these fixes address the mathematical functions. We would need to re-enable the MathJax over on the Learn platform for those to render properly. There's already a [PR](https://github.com/freeCodeCamp/learn/pull/244) to render mathematical functions only on the Rosetta Code pages that's waiting for QA/approval.

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
#### Description

Made a number of formatting changes to the challenge descriptions, and updated a number of the tests which were failing. More detail in https://github.com/freeCodeCamp/curriculum/issues/277.

#### Pre-Submission Checklist

- [x] Your pull request targets the `dev` branch.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/challenge-tests`)
- [x] All new and existing tests pass the command `npm test`.
- [x] Use `npm run commit` to generate a conventional commit message.
    Learn more here: <https://conventionalcommits.org/#why-use-conventional-commits>
- [x] The changes were done locally on your machine and NOT GitHub web interface.
    If they were done on the web interface you have ensured that you are creating conventional commit messages.

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #277

### CODE REVIEW

**Comment by NitronR** [comment]
@scissorsneedfoodtoo Sure!! I will take a look.

File: challenges/08-coding-interview-prep/rosetta-code.json [hunk 1]
```diff
@@ -5172,36 +5335,50 @@
...
+          "text": "<code>jortsort</code> should be a function.",
+          "testString":
+            "assert(typeof jortsort=='function','<code>jortsort</code> should be a function.');"
         },
         {
-          "text": "'<code>jortsort('+JSON.stringify(tests[0])+')</code> should return a boolean.'",
-          "testString": "assert(typeof jortsort(tests[0].slice())=='boolean','<code>jortsort('+JSON.stringify(tests[0])+')</code> should return a boolean.');"
+          "text": "<code>jortsort([1,2,3,4,5])</code> should return a boolean.",
+          "testString":
+            "assert(typeof jortsort([1,2,3,4,5].slice())=='boolean','<code>jortsort([1,2,3,4,5])</code> should return a boolean.');"
```
Comment (NitronR) [inline challenges/08-coding-interview-prep/rosetta-code.json]: `slice()` is not needed anymore as the array used is not stored in a variable. It was used earlier as the function would mutate the tests.

File: challenges/08-coding-interview-prep/rosetta-code.json [hunk 2]
```diff
@@ -5136,21 +5303,17 @@
...
-          "contents": [
-            "function jaro (s, t) {",
-            "  // Good luck!",
-            "}"
-          ],
+          "contents": ["function jaro (s, t) {", "  // Good luck!", "}"],
           "head": [],
           "tail": [
-            "let tests=[",
+            "let jaroDistanceTests = [",
```
Comment (NitronR) [inline challenges/08-coding-interview-prep/rosetta-code.json]: The jaroDistanceTests (previously tests) and results variables are not needed anymore as they were only used in the assert statements. These can be removed from the challenges where the assert statements have raw test values that are not stored in variables but are directly included.

**Review by NitronR — COMMENTED** [review]
@scissorsneedfoodtoo Great work!! There are a few suggestions below. Also if we can remove the `tests` and `results` variables in the tail, then maybe we can close [this](https://github.com/freeCodeCamp/learn/pull/271) PR?

File: challenges/08-coding-interview-prep/rosetta-code.json (hunk 1 above)
Comment (scissorsneedfoodtoo) [reply challenges/08-coding-interview-prep/rosetta-code.json]: Okay! I'll remove that and update the PR.

File: challenges/08-coding-interview-prep/rosetta-code.json (hunk 2 above)
Comment (scissorsneedfoodtoo) [reply challenges/08-coding-interview-prep/rosetta-code.json]: Good catch. I'll go through the challenges and clean up the unneeded test and results arrays where I can.

**Comment by scissorsneedfoodtoo** [comment]
@NitronR, I removed the `tests` and `results` variables from the tails of the problems I edited. There are still one or two of the problems where I had to leave them in because the solutions were very complicated or visual.

Would you mind taking another look when you get a chance?

**Comment by NitronR** [comment]
@scissorsneedfoodtoo I took a look just now. LGTM!!

**Comment by scissorsneedfoodtoo** [comment]
@NitronR, thank you a lot for your help!

**Comment by scissorsneedfoodtoo** [comment]
Though it's my own PR, I'll go ahead and merge it since it was checked out by @NitronR.

**Pull request merged**